app/packages/core/src/components/ColorModal/JSONViewer.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useRef } from "react";
+import { useRecoilState, useRecoilValue, useSetRecoilState } from "recoil";
+import Editor from "@monaco-editor/react";
+import * as fos from "@fiftyone/state";
+
+import { tempColorJSON, tempColorSetting, tempGlobalSetting } from "./utils";
+import { ActionOption } from "../Actions/Common";
+import { useTheme } from "@fiftyone/components";
+import { SORT_BY_SIMILARITY } from "../../utils/links";
+import { SectionWrapper } from "./ShareStyledDiv";
+
+const JSONViewer: React.FC = ({}) => {
+  const themeMode = useRecoilValue(fos.theme);
+  const theme = useTheme();
+  const editorRef = useRef(null);
+  const [global, setGlobal] = useRecoilState(tempGlobalSetting);
+  const setting = useRecoilValue(fos.sessionColorScheme);
+  const fullSetting = setting.customizedColorSettings ?? [];
+  const [data, setData] = useRecoilState(tempColorJSON);
+  const resetTempCustomizeColor = useSetRecoilState(tempColorSetting);
+  const colors = useRecoilValue(fos.coloring(false)).pool as string[];
+  const opacity = useRecoilValue(fos.alpha(false));
+  const colorBy = useRecoilValue(
+    fos.appConfigOption({ key: "colorBy", modal: false })
+  );
+  const useMulticolorKeypoints = useRecoilValue(
+    fos.appConfigOption({ key: "multicolorKeypoints", modal: false })
+  );
+  const showSkeleton = useRecoilValue(
+    fos.appConfigOption({ key: "showSkeletons", modal: false })
+  );
+
+  const handleEditorDidMount = (editor) => (editorRef.current = editor);
+  const handleEditorChange = (value: string | undefined) => {
+    value && setData(JSON.parse(value));
+  };
```

Comment:
nit: I noticed that across couple pages there are same code for gettingthese values/functions. You can abstract these into a single hook useCustomColors() that return all these values and functions. then you can use it like

const { colorBy, useMulticolorKeypoints, etc } = useCustomColors()